ExilenceNextApp/src/components/toolbar/Toolbar.tsx

```diff
@@ -266,6 +268,23 @@ const Toolbar = ({
                   </IconButton>
                 </span>
               </Tooltip>
+              <Tooltip title={t('label.remove_snapshot_icon_title') || ''} placement="bottom">
+                <span>
+                  <IconButton
+                    disabled={
+                      !activeProfile ||
+                      isSnapshotting ||
+                      !signalrOnline ||
+                      activeProfile.snapshots.length === 0
+                    }
```

Comment:
I suggest moving logic outside of props and just passing `isDisabled` variable

Reply:
Agreed, but for now this is fine IMO as its consistent with how the logic looks right now in the entire toolbar component. I think we have some old issue lying around related to refactoring this monster though 👹